include/fi_mem.h

```diff
@@ -88,11 +101,20 @@ union util_buf {
 	uint8_t data[0];
 };
 
+/* create buffer pool */
 struct util_buf_pool *util_buf_pool_create(size_t size, size_t alignment,
 					   size_t max_cnt, size_t chunk_cnt);
+
+/* create buffer pool with alloc/free handlers */
+struct util_buf_pool *util_buf_pool_ext_create(size_t size, size_t alignment,
+					    size_t max_cnt, size_t chunk_cnt,
+					    util_buf_region_alloc_hndlr alloc_hndlr,
+					    util_buf_region_free_hndlr free_hndlr);
 void util_buf_pool_destroy(struct util_buf_pool *pool);
 
 void *util_buf_get(struct util_buf_pool *pool);
+void *util_buf_ext_get(struct util_buf_pool *pool, void **context);
```

Comment:
I would move the 'ext' to the end of the function calls, and maybe just use 'ex'.